src/crypto/tls/ech_config.go

```diff
@@ -174,10 +158,35 @@ func (config *ECHConfig) isPeerCipherSuiteSupported(suite echCipherSuite) bool {
 // selectCipherSuite returns the first ciphersuite indicated by this
 // configuration that is supported by the caller.
 func (config *ECHConfig) selectCipherSuite() (echCipherSuite, error) {
-	for i := range config.suites {
-		if echIsCipherSuiteSupported(config.suites[i]) {
-			return config.suites[i], nil
+	for _, suite := range config.suites {
+		if hpke.KdfID(suite.kdfId).IsValid() &&
+			hpke.AeadID(suite.aeadId).IsValid() {
+			return suite, nil
 		}
 	}
 	return echCipherSuite{}, fmt.Errorf("could not negotiate a ciphersuite")
 }
+
+// id returns the configuration identifier for the given KDF.
+func (config *ECHConfig) id(kdfId hpke.KdfID) ([]byte, error) {
```

Comment:
This should use the HPKE KDF API directly, after [@armfazh's PR](https://github.com/armfazh/circl/pull/8) lands.

Reply:
Done.